twa

```diff
@@ -391,8 +392,78 @@ function stage_4_scm_repos {
 # * Fetch the server's cert using `openssl s_client`, check it for known bad suites
 #   * Alternatively, just `nmap --script ssl-enum-ciphers -p 443 $domain`
 
+function stage_5_CAA {
+  verbose "Stage 5: Certification Authority Authorization"
+  verbose "Which Certificate Authorities (CAs) are allowed to issue certificates for the domain"
+    code=`dig +short caa $domain`
+    number_of_lines=`echo $code|wc -l`
+    if [ $number_of_lines -gt 1 ];then
+      INFO "multiple CAAs specified, everyone join the party"
+      #Cloudflare currently set the following set of certs as CAA
+      if [ echo $code|grep -c 'issue “comodoca.com”' -gt 0 ];then
+        if [ echo $code|grep -c 'issue “digicert.com”' -gt 0 ];then
+          if [ echo $code|grep -c 'issue “globalsign.com”' -gt 0 ];then
+            if [ echo $code|grep -c 'issuewild “comodoca.com”' -gt 0 ];then
+              if [ echo $code|grep -c 'issuewild “digicert.com”' -gt 0 ];then
+                if [ echo $code|grep -c 'issuewild “globalsign.com”' -gt 0 ];then
+                  INFO "CAA is probably provided by Cloudflare"
+                fi
+              fi 
+            fi
+          fi
+        fi
+      fi
+    fi
+
+
+    echo $code| while read line ; do
```

Comment:
This and below is hard to read -- either use `read` directly, or use `awk` to extract the fields explicitly.